app/Models/tran.php

```diff
@@ -19,9 +20,31 @@ class tran extends Model
         'type',
         'status',
         'utr',
+        'invoice',
         'project',
         'comment',
         
     ];
 
+    public static function boot()
+    {
+        parent::boot();
+        static::creating(function($model){
+            $last_record = DB::table('trans')->latest()->first();
+            $entry_number = 0;
+            $zeros = 0;
```

Comment:
I don't think that these two variables are useful as they are simply adding a couple of zeros at the end of invoice number, so you can remove them.